packages/core/elasticsearch/core-elasticsearch-client-server-internal/src/create_transport.ts

```diff
@@ -41,7 +40,17 @@ export const createTransport = ({
 
     async request(params: TransportRequestParams, options?: TransportRequestOptions) {
       const opts: TransportRequestOptions = options ? { ...options } : {};
-      const opaqueId = getExecutionContext();
+      executionContext?.get()?.incrementCircuitBreaker('requests');
+      const requests = executionContext?.get()?.getCircuitBreaker('requests') ?? 0;
+      if (requests > 0 && requests % 100 === 0) {
```

Comment:
yeah I was logging every time requests > 100 which created way too much noise. But this can be ignored. The idea would be to increment until we hit the configured budget and then not increment anymore. We wouldn't "reset" the budget once hit.